pkg/admission/reservedmetadata/admission.go

```diff
@@ -84,67 +84,49 @@ func (o *reservedMetadata) Validate(ctx context.Context, a admission.Attributes,
 		return nil
 	}
 
-	if err := assertViolation(newMeta.GetAnnotations(), oldMeta.GetAnnotations(), annotationAllowList); err != nil {
-		return admission.NewForbidden(a, fmt.Errorf("modification of reserved annotation: %w", err))
+	if k, ok := isModified(newMeta.GetAnnotations(), oldMeta.GetAnnotations(), annotationAllowList); ok {
+		return admission.NewForbidden(a, fmt.Errorf("modification of reserved annotation: %q", k))
 	}
 
-	if err := assertViolation(newMeta.GetLabels(), oldMeta.GetLabels(), labelAllowList); err != nil {
-		return admission.NewForbidden(a, fmt.Errorf("modification of reserved label: %w", err))
+	if k, ok := isModified(newMeta.GetLabels(), oldMeta.GetLabels(), labelAllowList); ok {
+		return admission.NewForbidden(a, fmt.Errorf("modification of reserved label: %q", k))
 	}
 
 	return nil
 }
 
-// diff computes the difference between values of left and right.
-// It returns all keys from left and right, where:
-//
-// - left has a key not present in right
-// - right has a key not present in left
-// - the values differ between left and right for the same key
-func diff(left, right map[string]string) []string {
-	var (
-		result      = make([]string, 0, len(left)+len(right))
-		onlyOnRight = make(map[string]string, len(right))
-	)
-
-	for k, v := range right {
-		onlyOnRight[k] = v
+func isModified(new, old map[string]string, allowList []string) (key string, modified bool) {
```

Comment:
needs a better name. It is not only modified, but it's about privileged modification